OICI:213 - Formulaire OP3

Iandraina-etech opened the pull request:
Ajout des champs necessaire pour adaptation de l object claim a fin qu'il puisse prendre en compte les details de l'OP3

date preAuthorizatioon
code specifique a la preAuthorization
-status preAuth
-adaptation deschamps pour qu'il soient non necessaire dans la requette GQL mais qu'il y ai une verification cote back pour que le back (pour les claims classique) , ainsi la creation peut prendre en compte les OP3 et les PEC classiques
-Ajout de permissions pour le nouveau workflow
-MEP du nouveau workflow
-verification des prestation de PreAuth pour les services et items

Mirado-etech committed:
Add preauthorization logic

Iandraina-etech committed:
Merge pull request #28 from ArkeupIDComores/OICI-213-Mirado

Add preauthorization logic

Iandraina-etech committed:
Add pre-authorization code support to claims

Introduces code_pre_authorization and date_pre_authorization fields to the Claim model and related migration. Updates GraphQL schema, mutations, queries, and services to handle pre-authorization codes, including validation and uniqueness checks. Also fixes ICD field mapping in ClaimReportService.

Iandraina-etech committed:
Add pre-authorization support to claims

Introduces pre-authorization fields and logic to the Claim model, including status and rejection reason. Adds mutations, services, and validation for submitting and processing pre-authorization claims, as well as related GraphQL schema and query updates. Migration included for new model fields.

Iandraina-etech committed:
Add mutations for claim pre-authorization workflow

Introduces three new GraphQL mutations: submitClaimsToMedicalMutation, submitToNormalClaimMutation, and rejectClaimPreAuthorizationMutation. These mutations support submitting claims to a medical officer, validating claims, and rejecting claims with a reason, enhancing the claim pre-authorization process. Schema updated to expose these mutations.

Iandraina-etech committed:
Merge pull request #29 from ArkeupIDComores/OICI-213-Iandraina

Oici 213 iandraina

Iandraina-etech committed:
Enable claim history and status update on rejection

Uncommented calls to `save_history()` and `save()` in the `rejectClaimPreAuthorizationMutation` to ensure claim history is recorded and status is updated when a claim pre-authorization is rejected.

Iandraina-etech committed:
Merge pull request #30 from ArkeupIDComores/OICI-213-Iandraina

Enable claim history and status update on rejection

Iandraina-etech committed:
Add audit user tracking for claim pre-authorization

Introduces the audit_user_id_pre_auth field to the Claim model to track the user responsible for pre-authorization actions. Updates relevant mutations to set this field, adds new permissions for pre-authorization validation and rejection, and assigns these rights to medical officer and health facility admin roles via migrations.

Mirado-etech committed:
Normal claim filter

Add normal claim filter logic for claims ignoring pre authorizations.

Mirado-etech committed:
Merge pull request #31 from ArkeupIDComores/OICI-213-Mirado

Normal claim filter

Iandraina-etech committed:
Replace HF_ADMIN_ID with CLAIM_ADMIN in migration

Updated the migration to use CLAIM_ADMIN (256) instead of HF_ADMIN_ID (524288) when assigning and removing rights for claim admin roles. This change ensures consistency with the current role ID definitions.

Iandraina-etech committed:
Merge pull request #32 from ArkeupIDComores/OICI-213-Iandraina

Oici 213 iandraina

Iandraina-etech committed:
Add new permission checks for claim mutations

Introduced specific permission lists in ClaimConfig for validating admin HF pre-auth, medical pre-auth, and rejecting pre-auth. Updated corresponding GraphQL mutations to use these new permission checks instead of the previous submit claims permissions.

Iandraina-etech committed:
Merge pull request #33 from ArkeupIDComores/OICI-213

OICI 213 : Ajout formulaire OP3

kevel-dev commented on claim/gql_mutations.py:
```diff
@@ -231,18 +233,18 @@ class AttachmentInputType(Attachment, OpenIMISMutation.Input):
     id = graphene.Int(required=False, read_only=True)
     uuid = graphene.String(required=False)
-    code = ClaimCodeInputType(required=True)
+    code = ClaimCodeInputType(required=False)
```
Je peux voir le RFC qui dit que les champs: "code", "date_from", "date_claimed" sont requis uniquement s'il y'a pas préauthorisation ?

Iandraina-etech replied:
ces champs sont des champs de prestations classiques donc j'ai du modifier cux si pour que la prestation de preAuthorisaTion puisse etre insere meme si ces champs n'
existent pas
 voici le model op3 il n  y a pas de date from date claimed et le code est un code preAuthorisation
<img width="469" height="673" alt="image_2025-11-06_135346368" src="https://github.com/user-attachments/assets/0f177ca1-2bf7-46be-af71-1efbc030093e" />

kevel-dev replied:
Si ce n'est qu'a cause de ce rapport alors ce n'est pas normal. La solution la plus simple sera de mettre ce champ vide lors de la création du rapport meme si la valeur existe en BD. De plus je pense que ce rapport est juste pour présentation de la forme, il faudra remplir tout les champs présents lors de l'impression finale.

Iandraina-etech replied:
Donc on va mettre en required mais les champs cote fe renverront des champs vides ("") ?
est ce que ca ne revient pas  a la meme chose  vue que je verifie deja dans la mutation de creation que les champs ne soient pas None ou vide si ce n'est une preAuthorisation ?


ca risque meme de causer des erreurs car si jamais c'est required et j'envoie un "" alors que le type attendu est une date par exemple le GQL risque de raise une Erreur

kevel-dev replied:
Dans ce cas comment allez-vous identifier un claim de type pré authorisation parmi plusieurs si vous n'avez pas le code pour filtrer?

Iandraina-etech replied:
Il y a un code de preauthorisation et aussi un boolean is_pre_autorisation pour différencier on peut aussi filtrer avec le code pre authorization

kevel-dev commented on claim/gql_mutations.py:
```diff
@@ -360,7 +364,17 @@ def async_mutate(cls, user, **data):
+                    _("mutation.claims.missingfields"))
+            if is_pre_authorization==True and code_pre_authorization==None:
+                     raise ValidationError(
+                    _("mutation.claims.missingfields"))
```
Ici, le message renvoyé lorsque les champs sont manquants sera la meme que l'erreur levée lorsque le code_de_préauthorisation n'est pas renseigné. A mon sens ca risque embrouller l'utilisateur, il faut changer le message d'erreur dans le second cas.

Iandraina-etech replied:
resolu

kevel-dev replied:
Vous avez résolu comment ? je ne vois pas de changement.

Iandraina-etech replied:
on est entrait de commit les changementrs de fix

Iandraina-etech replied:
les changements on etet merge dans le dev comores mainenant

kevel-dev commented on claim/services.py:
```diff
@@ -543,7 +562,10 @@ def validate_claim_data(data, user):
+    if incoming_code!=None and len(incoming_code) > ClaimConfig.max_claim_length:
+        raise ValidationError(_("mutation.code_name_too_long"))
+    
+    if incoming_code_pre_authorization!=None and len(incoming_code_pre_authorization) > ClaimConfig.max_claim_length:
```
mutation.code_name_too_long renvoit au code qui est long, pour le nouveau champ code de préauthorisation il faut renovyer une erreur différente, ca risque embrouiller l'utilisateur.

Iandraina-etech replied:
Resolu

kevel-dev commented on claim/services.py:
```diff
@@ -399,6 +399,11 @@ def check_unique_claim_code(code):
     return []
 
+def check_unique_claim_pre_authorization_code(code_pre_authorization):
+    if Claim.objects.filter(code_pre_authorization=code_pre_authorization, validity_to__isnull=True).exists():
```
Les champs uniques doivent etre unique peu importe la validité

kevel-dev commented on claim/services.py:
```diff
@@ -306,10 +306,10 @@ def fetch(self, uuid):
-            "icd2": str(claim.icd1) if claim.icd_2 else None,
-            "icd3": str(claim.icd1) if claim.icd_3 else None,
-            "icd4": str(claim.icd1) if claim.icd_4 else None,
+            "icd1": str(claim.icd_1) if claim.icd_1 else None,
```
J'ai du mal a comprendre ce change, c'est possible d'avoir plus d'explication ?

Iandraina-etech replied:
J'ai remarque que l'ancien code levait une execption (field icd1 does not existe) donc j'ai du modifie cela en remplacant par les bon attributs

kevel-dev replied:
Non, sur nos instances nous n'avons pas ce probleme. C'est surement une erreur chez vous en local. le mieux c'est de laisser comme c'était.

Iandraina-etech replied:
Ok le print des infos du claim ne marchait pas en local du coup je l'ai modifie mais si vous le dites

Iandraina-etech replied:
<img width="1866" height="947" alt="image" src="https://github.com/user-attachments/assets/1aa6e9b7-ee9c-4f3b-802e-033c4a28f251" />
je parle du bouton en haut a droite du claim

Iandraina-etech replied:
<img width="1298" height="61" alt="image" src="https://github.com/user-attachments/assets/1e873210-2ae5-44cf-bfaf-b6c6a53b2319" />
voic le bouton tt a droite ca ne marchait pas avant que je modifie cette partie du code a cause de l'attribut icd1 mal tape ca devrait etre icd_1

kevel-dev commented on claim/migrations/0041_role_medical_officer.py:
```diff
@@ -0,0 +1,51 @@
+        insert_role_right_for_system(role, right)
+
+
+def _remove_rights_from_role(role,rightArray):
```
Pourquoi supprimer ces roles?

Iandraina-etech replied:
c'est  au cas ou on voudrait reverser la migration elle ne s'execute pas si on fait migrate
elle se base sur la migration 24 de claim que j'ai prix comme exemple pour l ajout des roles

Iandraina-etech replied:
<img width="1920" height="1008" alt="image_2025-11-06_140301339" src="https://github.com/user-attachments/assets/a3bd4c62-9f0a-4d19-b710-fe1d4ddd965d" />

Iandraina-etech committed:
Improve claim pre-authorization validation and tracking

Enhanced field-specific validation errors for claim creation, added date_pre_authorization_decision field to the Claim model and migration, and ensured this date is set on pre-authorization decision actions. Also updated error messages for code length validation to be more specific.

Iandraina-etech committed:
Add date_pre_authorization_emergency to Claim model

Introduces the date_pre_authorization_emergency field to the Claim model and GraphQL mutation input. Validation now requires this field for emergency visit types during pre-authorization. Migration updated to add the new field to the database.

Iandraina-etech committed:
Fix ICD field references and emergency visit type check

Corrects the mapping of ICD fields in ClaimReportService to use the appropriate claim attributes. Also refactors emergency visit type comparison in CreateClaimMutation to use a constant for consistency.

Iandraina-etech committed:
Merge pull request #34 from ArkeupIDComores/OICI-213

Oici 213
